src/webots/vrml/WbProtoManager.cpp

```diff
@@ -557,9 +559,13 @@ void WbProtoManager::generateProtoInfoMap(int category, bool regenerate) {
       }
 
       WbProtoInfo *info;
-      const bool isWebotsProto = isProtoInCategory(protoName, PROTO_WEBOTS) &&
-                                 (WbUrl::resolveUrl(protoUrl(protoName, PROTO_WEBOTS)) == WbUrl::resolveUrl(protoPath));
-      if (isCachedProto && isWebotsProto)
+      const bool isWebotsProto =
+        isProtoInCategory(protoName, PROTO_WEBOTS) &&
+        (WbUrl::resolveUrl(protoUrl(protoName, PROTO_WEBOTS)) ==
+         WbUrl::resolveUrl(isCachedProto ? WbNetwork::instance()->getUrlFromEphemeralCache(protoPath) : protoPath));
```

Comment:
since we go through the trouble of requesting the original URL beforehand to extract the PROTO name, I think we can store the URL in a variable at the same time to avoid calling `getUrlFromEphemeralCache` twice. Doing so also removes the need to use the ternary operator here, since by the point this line is reached, it's already either the correct path, or the reversed one